internals/workloads/workloads.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a Copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workloads
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/osutil"
+	"github.com/canonical/pebble/internals/plan"
+)
+
+var _ plan.SectionExtension = (*PlanExtension)(nil)
+
+type PlanExtension struct{}
+
+func (ext *PlanExtension) CombineSections(sections ...plan.Section) (plan.Section, error) {
+	ws := &WorkloadsSection{}
+	for _, section := range sections {
+		layer, ok := section.(*WorkloadsSection)
+		if !ok {
+			return nil, fmt.Errorf("internal error: invalid section type %T", layer)
+		}
+		if err := ws.combine(layer); err != nil {
+			return nil, err
+		}
+	}
+	return ws, nil
+}
+
+func (ext *PlanExtension) ParseSection(data yaml.Node) (plan.Section, error) {
+	ws := &WorkloadsSection{}
+	// The following issue prevents us from using the yaml.Node decoder
+	// with KnownFields = true behavior. Once one of the proposals get
+	// merged, we can remove the intermediate Marshal step.
+	// https://github.com/go-yaml/yaml/issues/460
+	if len(data.Content) != 0 {
+		if err := plan.SectionDecode(&data, ws); err != nil {
+			return nil, &plan.FormatError{
+				Message: fmt.Sprintf(`cannot parse the "workloads" section: %v`, err),
+			}
+		}
+	}
+	for name, workload := range ws.Entries {
+		if workload != nil {
```

Comment:
Is this condition ever false? There would be an entry in the `ws.Entries` map that has a `name` key that points to nothing?

Reply:
Yes, in the following case:

```yaml
workloads:
    default:
```